src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/CSharpFlagsEnumGenerator.cs

```diff
@@ -10,20 +10,20 @@ namespace Microsoft.CodeAnalysis.CSharp.CodeGeneration
 {
     internal class CSharpFlagsEnumGenerator : AbstractFlagsEnumGenerator
     {
-        internal static readonly CSharpFlagsEnumGenerator Instance = new();
-        private static readonly SyntaxGenerator s_generatorInstance = CSharpSyntaxGenerator.Instance;
```

Comment:
shared layer can't refer to this internal type.  So all these types now have the SyntaxGenerator passed into them.  it was a lot less impactful than i thought it would be.